diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/CollapseIfStatementsRule.kt

```diff
@@ -0,0 +1,31 @@
+package org.cqfn.diktat.ruleset.rules.chapter3
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.findChildrenMatching
+
+import com.pinterest.ktlint.core.ast.ElementType.IF
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtIfExpression
+
+class CollapseIfStatementsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "collapse-if",
+    configRules,
+    listOf(
+        Warnings.COLLAPSE_IF_STATEMENTS
+    )
+) {
+    override fun logic(node: ASTNode) {
+        if (node.elementType == IF) {
```

Comment:
if "if-block" contains comments before nested if - we should also fix it, but move the comment:
```
if (cond1) {
 //
  if (cond2) {

```

=> 
```
if (cond1 && 
  //
  cond2

```

Reply:
Corresponding issue https://github.com/cqfn/diKTat/issues/817